### PR TITLE
Discussion endpoints

### PR DESCRIPTION
**Pull request by alahirajeffrey**
- added migration files for discussion and comment tables
- added discussion and comments endpoint including endpoint to
1. start a discussion
2. get all discussions for an API
3. get a single discussion
4. add a parent comment under a discussion
5. add a child comment under a parent comment
6. edit a comment
7. get a parent comment and all its child comments
- updated idcheck guard to cover discussionId, commentId

### CODE REVIEW

File: src/discussion/discussion.controller.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    @Post('/add-child-comment/:profileId/:commentId')
+    @ApiOperation({summary:'Add a child comment to a parent comment'})
+    async addChildComment(
+        @Param('profileId') profileId: string,
+        @Param('commentId') commentId: string,
+        @Body() dto: CreateChildCommentDto
+    ){
+        const childComment = await this.discussionService.addChildComment(profileId, commentId, dto)
+        return ZaLaResponse.Ok(childComment, 'Ok', '201')
+    }
```
Comment (Chimexx) [inline src/discussion/discussion.controller.ts:83]: Are you supposed to go 3 steps deep? A post has a title and body, and that post has comments under it. I think it should stop  here

File: src/discussion/discussion.service.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                comments.push(childComment)
+            }
+            
+            return comments
+        } catch (error) {
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest(error.name, error.message, error.status)
+            )   
+        }
+    }
```
Comment (Chimexx) [inline src/discussion/discussion.service.ts:138]: I don't think this is efficient, if a comment should have comments under it, it should be another entity

**Review by Chimexx — CHANGES_REQUESTED** [review]
You are supposed to take the endpoints bunch by bunch, not the whole thing at once, Image yourself as a reviewer here.

File: src/discussion/discussion.controller.ts (hunk 1 above)
Comment (alahirajeffrey) [reply src/discussion/discussion.controller.ts:83]: I don't understand your comment. Can you please explain it a little bit more

File: src/discussion/discussion.service.ts (hunk 2 above)
Comment (alahirajeffrey) [reply src/discussion/discussion.service.ts:138]: Brain suggested it while we were discussing it. I'll bring it up to him during the standup this morning

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**